plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/plugins.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import interpolateComponents from '@automattic/interpolate-components';
+import { Link, Plugins as PluginInstaller } from '@woocommerce/components';
+import { OPTIONS_STORE_NAME, InstallPluginsResponse } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { Text } from '@woocommerce/experimental';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { createNoticesFromResponse } from '~/lib/notices';
+
+const isWcConnectOptions = (
+	wcConnectOptions: unknown
+): wcConnectOptions is {
+	[ key: string ]: unknown;
+} => typeof wcConnectOptions === 'object' && wcConnectOptions !== null;
+
+type Props = {
+	nextStep: () => void;
+	pluginsToActivate: string[];
+};
+
+export const Plugins: React.FC< Props > = ( {
+	nextStep,
+	pluginsToActivate,
+} ) => {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { isResolving, tosAccepted } = useSelect( ( select ) => {
+		const { getOption, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+		const wcConnectOptions = getOption( 'wc_connect_options' );
+
+		return {
+			isResolving:
+				! hasFinishedResolution( 'getOption', [
+					'woocommerce_setup_jetpack_opted_in',
+				] ) ||
+				! hasFinishedResolution( 'getOption', [
+					'wc_connect_options',
+				] ),
+			tosAccepted:
+				( isWcConnectOptions( wcConnectOptions ) &&
+					wcConnectOptions?.tos_accepted ) ||
+				getOption( 'woocommerce_setup_jetpack_opted_in' ) === '1',
+		};
+	} );
+
+	useEffect( () => {
+		if ( ! tosAccepted || pluginsToActivate.length ) {
+			return;
+		}
+
+		nextStep();
+	}, [ nextStep, pluginsToActivate, tosAccepted ] );
+	const agreementText = pluginsToActivate.includes( 'woocommerce-services' )
+		? __(
+				'By installing Jetpack and WooCommerce Shipping you agree to the {{link}}Terms of Service{{/link}}.',
+				'woocommerce'
+		  )
+		: __(
+				'By installing Jetpack you agree to the {{link}}Terms of Service{{/link}}.',
+				'woocommerce'
+		  );
+
+	if ( isResolving ) {
+		return null;
+	}
+
+	return (
+		<>
+			<PluginInstaller
```

Comment:
I think this is replicated from the tax component, but I find it very confusing when we're not providing plugin slugs. 

I didn't know how it works until I checked the `PluginInstaller` code. 

https://github.com/woocommerce/woocommerce/blob/e9006b56dfc9747a23a5f8b0796b4a7e0576e14a/packages/js/components/src/plugins/index.js#L175-L180

Would you like to explicitly set the `pluginSlugs` prop?

Reply:
That's an excellent suggestion, I do think the implicit use of the default value is unreliable and less readable.